stonesoup/models/transition/linear.py

```diff
@@ -501,7 +380,147 @@ def constant_derivative(self):
         return 2
 
 
-class Singer(LinearGaussianTransitionModel, TimeVariantModel):
+class NthDerivativeDecay(LinearGaussianTransitionModel, TimeVariantModel):
+    r"""Model based on the Nth derivative with respect to time decaying to 0
+    exponentially, to set derivative use keyword argument
+    :attr:`decay_derivative`
+
+        The model is described by the following SDEs:
+
+        .. math::
+            :nowrap:
+
+            \begin{eqnarray}
+                dx^{(N-1)} & = & x^{(N)} dt & | {(N-1)th derivative \ on \
+                X-axis (m)} \\
+                dx^{(N)} & = & -K x^{N} dt + q\cdot dW_t,\ W_t \sim
+                \mathcal{N}(0,q^2) & | \ Nth\ derivative\ on\ X-axis (m/s^{N})
+            \end{eqnarray}
+
+    The transition and covarience matrices are very difficult to express
+    simply, but examples for N=1 and N=2 are given in
+    :class:`~.OrnsteinUhlenbeck` and :class:`~.Singer` respectively.
+        """
+    decay_derivative = Property(
+        int, doc="The derivative with respect to time to decay exponentially, "
+                 "eg if 2 identical to singer")
+    noise_diff_coeff = Property(
+        float, doc="The noise diffusion coefficient :math:`q`")
+    damping_coeff = Property(
+        float, doc="The Nth derivative damping coefficient :math:`K`")
+
+    @property
+    def ndim_state(self):
+        return self.decay_derivative + 1
+
+    @staticmethod
+    @lru_cache()
```

Comment:
Like the use of `lrucache`. Have you considered whether an explicit cache size is useful? Is the default of 128 sensible given the likely number of combinations of `N` and `K`?